client/src/partials/reports/global_transaction/global_transaction.js

```diff
@@ -133,45 +142,64 @@ angular.module('bhima.controllers')
     }
 
     function search () {
-      if (!$scope.model.account_id) { return; }
-      $scope.mode = $scope.model.account_id !== 0 ? 'selected' : 'all';
+      $scope.mode = ($scope.model.account_id && $scope.model.account_id > 0) ? 'selected' : 'all';
+      
+      if ($scope.model.account_id && $scope.model.account_id > 0) {
+        $scope.model.account_number = $scope.accounts.get($scope.model.account_id).account_number;       
+      } else {
+        $scope.model.account_number = 0;
+      }
+
       var qo = {
         source : $scope.model.source_id,
         enterprise_id : $scope.enterprise.id,
         account_id : $scope.model.account_id,
-        datef : util.sqlDate($scope.state.from),
-        datet : util.sqlDate($scope.state.to)
+        datef : util.sqlDate($scope.dates.from),
+        datet : util.sqlDate($scope.dates.to)
       };
 
-      if ($scope.model.account_id && $scope.model.account_id === 0) {
-        $scope.model.account_number = 'Tous';
-      } else {
-        $scope.model.account_number = $scope.accounts.get($scope.model.account_id).account_number;
-      }
-
       connect.fetch('/reports/allTrans/?'+JSON.stringify(qo))
       .then(function (res) {
         if (res.length > 0) {
           res.map(function (item) {
-            item.debit = getValue(map[util.sqlDate(item.trans_date)], item.debit, $scope.enterprise.currency_id);
-            item.credit = getValue(map[util.sqlDate(item.trans_date)], item.credit, $scope.enterprise.currency_id);
+              if($scope.enterprise.currency_id !== item.currency_id){
+                item.debit *= exchange.rate(item.debit,item.currency_id,new Date());
+                item.credit *= exchange.rate(item.debit,item.currency_id,new Date()); 
+              }
           });
           $scope.records = res;
           getTotal(res);
         } else {
-          getTotal(res);
+          $scope.somCredit = 0;
+          $scope.somDebit = 0;
           $scope.records = [];
         }
       });
     }
 
     function getTotal(items) {
+      var sCredit = 0, 
+        sDebit = 0;
       $scope.somCredit = 0;
       $scope.somDebit = 0;
       if (items.length > 0) {
```

Comment:
Just like above, you do not need to check `items.length > 0` before calling items.forEach().  If items is an empty array, .forEach() will not run.

If you are running into errors where items in undefined, you should check for that instead.  Use `angular.isDefined(items)` as your check.